File: all ideas/calculator.cpp
#include<iostream>
#include<cmath>

using namespace std;

void arithmetic(){
    int op = 0;
    float A = 0;
    float B = 0;

    cout << "Select operation\n";
    cout << "[1] Addition\n";
    cout << "[2] Subtraction\n";
    cout << "[3] Product\n";
    cout << "[4] Division\n";

    cin >> op;
    cout << "Enter the first number: ";
    cin >> A;
    cout << "Enter the second number: ";
    cin >> B;

    cout << "Result: ";

    switch(op){
        case 1:
            cout << (A + B);
            break;
        case 2:
            cout << (A - B);
            break;
        case 3:
            cout << (A * B);
            break;
        case 4:
            cout << (A / B);
            break;
        default:
            cout << "Invalid operation";
            break;
    }
    cout << endl;
}

void trigonometric(){
    int op = 0;
    float val = 0.0;

    cout << "Select\n";
    cout << "[1] Sine\n";
    cout << "[2] Cosine\n";
    cout << "Op: ";
    cin >> op;
    cout << "Enter the value: ";
    cin >> val;

    if(op == 1){
        cout << sin(val);
    }
    else if(op == 2){
        cout << cos(val);
    }
    else{
        cout << "Invalid operation";
    }
    cout << endl;
}

void exponential(){
    float base = 0.0;
    float eee = 0.0;

    cout << "Enter the base: ";
    cin >> base;
    cout << "Enter the exponent: ";
    cin >> eee;

    cout << pow(base, eee) << endl;
}

void logarithmic(){
    float value = 0.0;
    cout << "Enter the value to calculate the natural logarithm (log(e)): ";
    cin >> value;
    cout << log(value) << endl;
}

int main(){
    int sel = 0;
    char choice = 'y';
    cout << "Advanced Calculator\n";

    cout << "[1] Arithmetic\n";
    cout << "[2] Trigonometric\n";
    cout << "[3] Exponential\n";
    cout << "[4] Logarithmic\n";
    cout << "Your choice: ";

    while(choice == 'y'){
        cout << "Enter the type of operation you want to calculate: ";
        cin >> sel;

        switch(sel){
            case 1:
                arithmetic();
                break;
            case 2:
                trigonometric();
                break;
            case 3:
                exponential();
                break;
            case 4:
                logarithmic();
                break;
            default:
                cout << "Invalid Operation";
        }

        cout << "Do you want to continue? (y/n)" << endl;
        cin >> choice;
        if(choice == 'n'){
            break;
        }
    }

    return 0;
}

File: c#/calculator.cs
using System;

class calculator
{
    static void Main()
    {
        bool exit = false;
        while (!exit)
        {
            Console.WriteLine("Calculator");
            Console.WriteLine("1. Addition");
            Console.WriteLine("2. Subtraction");
            Console.WriteLine("3. Multiplication");
            Console.WriteLine("4. Division");
            Console.WriteLine("5. Exit");
            Console.Write("Enter your choice (1-5): ");

            string choice = Console.ReadLine();
            Console.WriteLine();

            switch (choice)
            {
                case "1":
                    Addition();
                    break;
                case "2":
                    Subtraction();
                    break;
                case "3":
                    Multiplication();
                    break;
                case "4":
                    Division();
                    break;
                case "5":
                    exit = true;
                    Console.WriteLine("Exiting...");
                    break;
                default:
                    Console.WriteLine("Invalid choice. Please try again.");
                    break;
            }

            Console.WriteLine();
        }
    }

    static void Addition()
    {
        Console.WriteLine("Addition");
        Console.Write("Enter the first number: ");
        double num1 = Convert.ToDouble(Console.ReadLine());
        Console.Write("Enter the second number: ");
        double num2 = Convert.ToDouble(Console.ReadLine());
        double result = num1 + num2;
        Console.WriteLine($"Result: {result}");
    }

    static void Subtraction()
    {
        Console.WriteLine("Subtraction");
        Console.Write("Enter the first number: ");
        double num1 = Convert.ToDouble(Console.ReadLine());
        Console.Write("Enter the second number: ");
        double num2 = Convert.ToDouble(Console.ReadLine());
        double result = num1 - num2;
        Console.WriteLine($"Result: {result}");
    }

    static void Multiplication()
    {
        Console.WriteLine("Multiplication");
        Console.Write("Enter the first number: ");
        double num1 = Convert.ToDouble(Console.ReadLine());
        Console.Write("Enter the second number: ");
        double num2 = Convert.ToDouble(Console.ReadLine());
        double result = num1 * num2;
        Console.WriteLine($"Result: {result}");
    }

    static void Division()
    {
        Console.WriteLine("Division");
        Console.Write("Enter the dividend: ");
        double dividend = Convert.ToDouble(Console.ReadLine());
        Console.Write("Enter the divisor: ");
        double divisor = Convert.ToDouble(Console.ReadLine());

        if (divisor == 0)
        {
            Console.WriteLine("Error: Cannot divide by zero.");
        }
        else
        {
            double result = dividend / divisor;
            Console.WriteLine($"Result: {result}");
        }
    }
}
